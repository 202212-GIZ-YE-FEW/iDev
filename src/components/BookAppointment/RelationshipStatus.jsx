import { withTranslation } from "next-i18next";

import RadioGroup from "../ui/radiogroup/RadioGroup";
import RadioInputItem from "../ui/radiogroup/RadioInputItem";

const RelationshipStatus = ({ t, name, relationshipStatus, onChange }) => {
    return (
        <>
            <RadioGroup title={t("whatRelationshipStatus")} asButton={true}>
                <RadioInputItem
                    id='single'
                    name={name}
                    value='single'
                    title={t("single")}
                    checked={relationshipStatus === "single"}
                    onChange={(e) => onChange(e)}
                />
                <RadioInputItem
                    id='married'
                    name={name}
                    value='married'
                    title={t("married")}
                    checked={relationshipStatus === "married"}
                    onChange={(e) => onChange(e)}
                />
                <RadioInputItem
                    id='divorced'
                    name={name}
                    value='divorced'
                    title={t("divorced")}
                    checked={relationshipStatus === "divorced"}
                    onChange={(e) => onChange(e)}
                />
            </RadioGroup>
        </>
    );
};

export default withTranslation("appointment")(RelationshipStatus);
