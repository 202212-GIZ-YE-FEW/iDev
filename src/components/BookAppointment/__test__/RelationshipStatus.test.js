import renderer from "react-test-renderer";

import RelationshipStatus from "../RelationshipStatus";

it("renders correctly", () => {
    const tree = renderer.create(<RelationshipStatus />).toJSON();
    expect(tree).toMatchSnapshot();
});
