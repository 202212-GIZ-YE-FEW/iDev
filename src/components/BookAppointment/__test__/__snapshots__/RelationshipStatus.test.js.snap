// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <p
    className="mb-4 p-2 text:lg md:text-2xl lg:text-3xl font-normal break-words text-black"
  >
    whatRelationshipStatus
  </p>,
  <fieldset
    className="space-y-4"
  >
    <div
      className="text-9xl font-normal text-black"
    >
      <div
        className="flex items-center space-y-4"
      >
        <input
          checked={false}
          className="peer hidden"
          id="single"
          onChange={[Function]}
          type="radio"
          value="single"
        />
        <label
          className="w-full flex cursor-pointer rounded-md p-4 text-base md:text-xl lg:text-2xl shadow-md border-[1px] border-light-gray/80 hover:border-gray peer-checked:border-none peer-checked:bg-cyan peer-checked:text-white"
          htmlFor="single"
        >
          single
           
        </label>
      </div>
      <div
        className="flex items-center space-y-4"
      >
        <input
          checked={false}
          className="peer hidden"
          id="married"
          onChange={[Function]}
          type="radio"
          value="married"
        />
        <label
          className="w-full flex cursor-pointer rounded-md p-4 text-base md:text-xl lg:text-2xl shadow-md border-[1px] border-light-gray/80 hover:border-gray peer-checked:border-none peer-checked:bg-cyan peer-checked:text-white"
          htmlFor="married"
        >
          married
           
        </label>
      </div>
      <div
        className="flex items-center space-y-4"
      >
        <input
          checked={false}
          className="peer hidden"
          id="divorced"
          onChange={[Function]}
          type="radio"
          value="divorced"
        />
        <label
          className="w-full flex cursor-pointer rounded-md p-4 text-base md:text-xl lg:text-2xl shadow-md border-[1px] border-light-gray/80 hover:border-gray peer-checked:border-none peer-checked:bg-cyan peer-checked:text-white"
          htmlFor="divorced"
        >
          divorced
           
        </label>
      </div>
    </div>
  </fieldset>,
]
`;
